CheapShopWeb/Services/Filtering.cs

```diff
@@ -51,28 +54,101 @@ public static List<Product> Filter(List<Product> productList, string name, strin
             return productList;
         }
 
-        public static List<Product> GetSimilarProducts(List<Product> productList, Product prod)
+        public static List<Product> GetSimilarProducts(List<Product> productList, Product prod, string searchString)
         {
 
-            //nezinau ka darau
+            productList = productList.FindAll(product => searchString.ToLower().Split(' ').All(query => product.name.ToLower().Contains(query)));
             List<Product> newlist= new List<Product>();
-            newlist.Add(prod);
-            string[] searchString = prod.name.Split(' ');
-            if (!string.IsNullOrEmpty(prod.name))
+            foreach (var product in productList)
             {
-                foreach (var product in productList)
+                if (LevenshteinDistance.Compute(product.name, prod.name) <= 3*prod.name.Split(' ').Length) //or 4*//
                 {
-                    
+                    newlist.Add(product);
                 }
             }
 
+            newlist.Sort((x, y) => String.Compare(x.price, y.price, StringComparison.Ordinal));
+            newlist.Insert(0, prod);
             return newlist;
         }
-        
-        public static Func<string, float> SToFFunc = num =>  //string to float function for lambda expression // makes number string comparable
+        public static List<Product> GetSimilarProductsGroup(List<Product> productList, Product prod, string itemsGroup)
         {
-            float n = float.Parse(num) * 100;
-            return n;
-        };
+
+            var method = typeof(SmallerGroups).GetMethod(itemsGroup + "Group");
+            if (!(method == null))
+            {
+                var smallerGroupList = (List<string>)method.Invoke(new SmallerGroups(), null); 
+                productList = productList.FindAll(product =>
+                        smallerGroupList.Any(group => product.group.ToLower().Equals(group.ToLower())));
+            } 
+            List<Product> newlist = new List<Product>();
+            foreach (var product in productList)
+            {
+                if (LevenshteinDistance.Compute(product.name, prod.name) <= 3 * prod.name.Split(' ').Length) //or 4*//
+                {
+                    newlist.Add(product);
+                }
+            }
+            newlist.Sort((x, y) => String.Compare(x.price, y.price, StringComparison.Ordinal));
+            newlist.Insert(0, prod);
+            return newlist;
+        }
+
+        public static Func<string, float> SToFFunc =
+            num => //string to float function for lambda expression // makes number string comparable
+            {
+                float n = float.Parse(num) * 100;
+                return n;
+            };
+    }
+    static class LevenshteinDistance
```

Comment:
Butu idomu suzinoti kaip sis algoritmas veikia

Reply:
Paprastas paaiškinimas:
"For example, ant and aunt are two different words. However, not as different as ant and antidote. The “edit distance” between ant and aunt is smaller than the “edit distance” between ant and antidote.  By edit distance, I mean the number of edits (removals, inserts, and replacements) needed to turn one string into another."